src-tauri/src/analytics/algos_config.rs

```diff
@@ -23,7 +23,7 @@ impl Params {
         }
     }
 
-    pub fn add_param(&mut self, key: String, value: Box<dyn Any>) {
+    pub fn add_param(&mut self, key: String, value: Box<dyn Any + Sync + Send>) {
```

Comment:
Why are these parameters `Any`? Is there not a way to use an `enum` so that we don't have to manually specify `Send + Sync`?

Reply:
The values of the hashmap are of type `Box<dyn Any>`. `Box` provides a pointer for (dynamic) heap allocation. Because all threads share the heap, this memory is not inherently threadsafe. 

Therefore, to make it threadsafe (and allow us to add a mutex/ARC to a `State` struct), I added the additional specifiers `Send` and `Sync` to the function definition. This isn't an enum - it means that any input of type `Any` must also implement `Send` and `Sync`. 
https://stackoverflow.com/questions/25649423/sending-trait-objects-between-threads-in-rust

Reply:
Now, as to why we're using heap memory in the first place, I have no idea. @barkincavdaroglu, any chance you remember why?

Reply:
We have to hardcode / know beforehand all the possible param types to use enum, which I didn't at the time. This was also easier to implement.

Reply:
Feel free to change it though

Reply:
Changing this implementation feels out of scope for this PR, because it involves a narrow refactor of code that was designed to be general. Because the current implementation works and makes sense, I'm leaving it as is.